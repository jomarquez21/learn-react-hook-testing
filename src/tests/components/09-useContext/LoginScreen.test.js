import React from 'react';
import {mount, shallow} from 'enzyme';
import {LoginScreen} from './../../../components/09-useContext/LoginScreen';
import {UserContext} from './../../../components/09-useContext/UserContext';

describe('Test on `LoginScreen`', () => {
  const setUser = jest.fn();

  const component = mount(
    <UserContext.Provider value={{setUser}}>
      <LoginScreen />
    </UserContext.Provider>
  );

  test('Debe renderizar correctamente', () => {
    expect(component).toMatchSnapshot();
  });

  test('Debe ejecutar la function `setUser`', () => {
    component.find('button').simulate('click');

    expect(setUser).toHaveBeenCalledTimes(1);

    const user = {
      id: 123,
      name: 'Jomarquez',
    };

    expect(setUser).toHaveBeenCalledWith(user);
  });
});
