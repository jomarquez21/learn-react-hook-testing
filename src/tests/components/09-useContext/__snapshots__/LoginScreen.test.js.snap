// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on \`LoginScreen\` Debe renderizar correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
